provider/src/test/kotlin/org/gradle/kotlin/dsl/accessors/ProjectSchemaAccessorsIntegrationTest.kt

```diff
@@ -436,6 +436,31 @@ class ProjectSchemaAccessorsIntegrationTest : AbstractIntegrationTest() {
         build("help")
     }
 
+    @Test
+    fun `can access source sets conventions registered by declared plugins via jit accessors`() {
+
+        withBuildScript("""
+            plugins {
+                groovy
+            }
+
+            java.sourceSets["main"].groovy {
+                groovy.srcDir("some/path")
+            }
+
+            val main by java.sourceSets
+            main.groovy.groovy {
+                srcDir("another/path")
+            }
+
+            val configured = main.groovy.groovy.srcDirs
+            val expected = linkedSetOf(file("src/main/groovy"), file("some/path"), file("another/path"))
+            require(configured == expected)
+        """)
+
+        build("help")
+    }
```

Comment:
Should you also add a test for the Kotlin plugin as well?

Reply:
It is also already covered in `GenerateProjectSchemaTest`

Reply:
But integration coverage for a non-core plugin adding `SourceSet` conventions to account for the different classloading at play should indeed be added.